src/components/network/line-layer.js

```diff
@@ -195,336 +206,808 @@ class LineLayer extends CompositeLayer {
+                        },
+                    })
+                )
+        );
+        this.setState({ layers: layers });
+        let next = undefined;
+        if (
+            doNextNominalVoltage &&
+            this.hasNextNominalVoltage(nominalVoltageIndex)
+        )
+            next = (ts) =>
+                this.computeArrows(
+                    ts,
+                    nominalVoltageIndex + 1,
+                    doLabel,
+                    doNextNominalVoltage
                 );
-                line.proximityFactorEnd = this.getProximityFactor(
-                    positions[positions.length - 2],
-                    positions[positions.length - 1]
+        else if (doLabel)
+            next = (ts) => this.computeLabels(ts, 0, doNextNominalVoltage);
+        if (next !== undefined) this.scheduleNext(timestamp, next);
+    }
+
+    generateLabelLayer(timestamp, nominalVoltageIndex, doNextNominalVoltage) {
+        const nominalVoltage = this.props.network.nominalVoltages[
+            nominalVoltageIndex
+        ];
+        let layers = this.state.layers;
+        const idLayer = 'ActivePower' + nominalVoltage;
+        layers.set(
+            idLayer,
+            (state, props) =>
+                new TextLayer(
+                    this.getSubLayerProps({
+                        id: idLayer,
+                        data: state.activePower.get(nominalVoltageIndex),
+                        getText: (activePower) =>
+                            activePower.p !== undefined
+                                ? Math.round(activePower.p).toString()
+                                : '',
+                        getPosition: (activePower) => activePower.printPosition,
+                        getColor: props.labelColor,
+                        fontFamily: 'Roboto',
+                        getSize: props.labelSize,
+                        getAngle: 0,
+                        getPixelOffset: (activePower) => activePower.offset,
+                        getTextAnchor: 'middle',
+                        visible:
+                            props.filteredNominalVoltages.includes(
+                                nominalVoltage
+                            ) && props.labelsVisible,
+                        updateTriggers: {
+                            getPosition: [
+                                state[ACTIVE_POWER + nominalVoltageIndex],
+                            ],
+                            getPixelOffset: [
+                                state[ACTIVE_POWER + nominalVoltageIndex],
+                            ],
+                            getText: [
+                                state[ACTIVE_POWER + nominalVoltageIndex],
+                            ],
+                            data: [state[ACTIVE_POWER + nominalVoltageIndex]],
+                        },
+                    })
+                )
+        );
+        this.setState({ layers: layers }, () => this.scheduleNext(next));
```

Comment:
same question, when is the lambda executed and what is the value of next?

Reply:
removed